### PR TITLE
8299500: Usage of constructors of primitive wrapper classes should be avoided in java.text API docs

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Removed constructors of primitive wrapper classes (deprecated for removal) for the following
- _java.text.ChoiceFormat_
- _java.text.MessageFormat_ 

Replaced with .valueOf() method

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8299500](https://bugs.openjdk.org/browse/JDK-8299500): Usage of constructors of primitive wrapper classes should be avoided in java.text API docs

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to [d6332659](https://git.openjdk.org/jdk/pull/11884/files/d63326594799a0226b413cd141852b3ff6989276)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11884/head:pull/11884` \
`$ git checkout pull/11884`

Update a local copy of the PR: \
`$ git checkout pull/11884` \
`$ git pull https://git.openjdk.org/jdk pull/11884/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11884`

View PR using the GUI difftool: \
`$ git pr show -t 11884`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11884.diff">https://git.openjdk.org/jdk/pull/11884.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/text/MessageFormat.java [hunk 1]
```diff
@@ -307,12 +307,12 @@
  * will be the final result of the parsing.  For example,
  * <blockquote><pre>
  * MessageFormat mf = new MessageFormat("{0,number,#.##}, {0,number,#.#}");
- * Object[] objs = {new Double(3.1415)};
+ * Object[] objs = {Double.valueOf(3.1415)};
  * String result = mf.format( objs );
  * // result now equals "3.14, 3.1"
  * objs = null;
```
Comment (naotoj) [inline src/java.base/share/classes/java/text/MessageFormat.java:313]: Although this is not related to this fix, this line is not needed (and is confusing as an example snippet). I'd remove this.

**Review by naotoj — APPROVED** [review]
LGTM

**Review by LanceAndersen — APPROVED** [review]
Looks fine Justin, thanks for doing this.

At some point we should convert the examples to be a snippet which we can do at a later time

**Comment by justin-curtis-lu** [comment]
> Looks fine Justin, thanks for doing this.
> 
> At some point we should convert the examples to be a snippet which we can do at a later time

Sounds good 👍
